### PR TITLE
Fix error messages

### LINKED ISSUE #492: [PE-D] Incorrect Data Format
I have copied the example provided in the user guide 
eg. add -x POST -u https://reqres.in/api/users -d {“name”: “tarzan”, “job”: “the jungle man”} -h “Content-Type: application/json” -t nature
but i keep getting the error that "Headers should be in the "key: Value" format. Perhaps there is a problem with the header and data parsing since the data seems to be in the correct format.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: tashawan23/ped#1

### LINKED ISSUE #561: Fix error messages in app and make formatting less cluttered

### LINKED ISSUE #567: Enhancement to error messages
If not index if provided for commands EDIT, SHOW, REMOVE, SEND, should we make an error message that says "no index provided" if there is no index provided

E.G, user just types in `show`

currently, all of it says: "Error: Index provided is not a non-zero unsigned integer within the allowed range.", which can be misinterpreted

### PR DESCRIPTION
**Pull request by JulietTeoh**
Closes #567 
Closes #561

@tlylt can u help me check if this is the intended UI output, especially the index errors?

show valid
Screenshot 2021-04-06 at 11 31 48 PM

edit valid
Screenshot 2021-04-07 at 12 58 25 AM

add valid
Screenshot 2021-04-07 at 12 57 58 AM

ERROR MESSAGE BELOW:

add
Screenshot 2021-04-07 at 12 49 00 AM

edit
Screenshot 2021-04-07 at 12 50 08 AM

run
Screenshot 2021-04-07 at 12 50 26 AM

find 
Screenshot 2021-04-07 at 12 50 40 AM

edit index = 0
Screenshot 2021-04-07 at 12 51 06 AM

edit index = -1000
Screenshot 2021-04-07 at 12 51 23 AM

edit index = 20
Screenshot 2021-04-07 at 12 51 49 AM

edit index = 10000000000
Screenshot 2021-04-07 at 12 52 01 AM

### CODE REVIEW

**Comment by JulietTeoh** [comment]
Updated valid index error message too

File: src/main/java/seedu/us/among/model/endpoint/Response.java [hunk 1]
```diff
@@ -122,7 +122,7 @@ public String toString() {
                 + getStatusCode()
                 + " | Reason Phrase: "
                 + getReasonPhrase()
-                + "\nResponse Body:\n"
+                + "\n\nResponse Body:\n"
                 + getResponseEntity();
     }
```
Comment (ong6) [inline src/main/java/seedu/us/among/model/endpoint/Response.java:127]: i think can remove these after warren does his addition

**Review by ong6 — COMMENTED** [review]
It looks ok to me, will let @tlylt approve it though.

**Review by tlylt — COMMENTED** [review]
Great changes, just two minor comments:
- the phrasing might make users think that INDEX is an optional parameter?
image
- for the other invalid command format errors, should we still keep the [ ]? because [ ] has the meaning of Repeated parameters in UG, not sure if this is still what we intended to mean in the error message

**Comment by JulietTeoh** [comment]
> Great changes, just two minor comments:
> 
> * the phrasing might make users think that INDEX is an optional parameter?
>   image
> * for the other invalid command format errors, should we still keep the [ ]? because [ ] has the meaning of Repeated parameters in UG, not sure if this is still what we intended to mean in the error message

I think [] is the best way to express that a parameter is repeated w/o writing a lot of words

**Review by tlylt — APPROVED** [review]
LGTM

**Pull request merged**